webapp/src/main/java/com/box/l10n/mojito/service/tm/GenerateMultiLocalizedAssetJob.java

```diff
@@ -0,0 +1,60 @@
+package com.box.l10n.mojito.service.tm;
+
+import com.box.l10n.mojito.quartz.QuartzJobInfo;
+import com.box.l10n.mojito.quartz.QuartzPollableJob;
+import com.box.l10n.mojito.quartz.QuartzPollableTaskScheduler;
+import com.box.l10n.mojito.rest.asset.LocaleInfo;
+import com.box.l10n.mojito.rest.asset.LocalizedAssetBody;
+import com.box.l10n.mojito.rest.asset.MultiLocalizedAssetBody;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class GenerateMultiLocalizedAssetJob
+    extends QuartzPollableJob<MultiLocalizedAssetBody, MultiLocalizedAssetBody> {
+
+  @Autowired QuartzPollableTaskScheduler quartzPollableTaskScheduler;
+
+  @Override
+  public MultiLocalizedAssetBody call(MultiLocalizedAssetBody input) throws Exception {
+
+    for (LocaleInfo localeInfo : input.getLocaleInfos()) {
+      String outputTag =
+          localeInfo.getOutputBcp47tag() != null
+              ? localeInfo.getOutputBcp47tag()
+              : localeInfo.getBcp47Tag();
+      QuartzJobInfo<LocalizedAssetBody, LocalizedAssetBody> quartzJobInfo =
+          QuartzJobInfo.newBuilder(GenerateLocalizedAssetJob.class)
+              .withInlineInput(false)
+              .withParentId(getParentId())
+              .withInput(createLocalizedAssetBody(localeInfo, input))
```

Comment:
thinking out loud... So we duplicate the source for each job, vs having the source saved once and read by multiple jobs? I guess it duplicates the data but the jobs are fully isolated. Any thoughts on that in general?

Reply:
Doing it this way using isolated jobs means we can re-use the existing single locale generation logic without updates meaning we avoid introducing regressions into the existing flows. 

Using this approach we only send the source from the client to the server once where previously we'd send it with every individual locale, I felt re-using the existing generation locale asset logic was a better fit for 2 reasons:

1. We know it works 😄 
2. Easier maintenance going forward as pull, async-ws & parallel all use the same code for individual locale asset generation.